app/models/project_services/irker_service.rb

```diff
@@ -0,0 +1,153 @@
+      "administrator: #{default_irc}. You can thus just give a channel name."
+    end
+    msg
+  end
+
+  def to_param
+    'irker'
+  end
+
+  def execute(push_data)
+    IrkerWorker.perform_async(project_id, channels,
+        colorize_messages, push_data)
+  end
+
+  def fields
+    [
+      { type: 'textarea', name: 'recipients',
+          placeholder: 'Recipients/channels separated by whitespaces' },
+      { type: 'checkbox', name: 'colorize_messages' },
+    ]
+  end
+
+  def check_irker
+    begin
+      host = Gitlab.config.irker.server_ip
+      port = Gitlab.config.irker.server_port
+    rescue Settingslogic::MissingSetting
+      host = 'localhost'
+      port = 6659
+    end
+    begin
+      TCPSocket.new(host, port).close
+    rescue Errno::ECONNREFUSED => e
+      logger.fatal "Can't connect to Irker daemon: #{e}"
+      msg  = "- Can't connect to the Irker daemon, "
+      msg += "please contact the server administrator if the problem persists."
+      errors.add(:active, msg)
+    rescue Exception => e
+      logger.fatal "Error while connecting to the Irker daemon: #{e}"
+      msg  = "- Error, please contact the server administrator "
+      msg += "if the problem persists."
+      errors.add(:active, msg)
+    end
+  end
+
+  def get_channels
+    return true unless :activated?
+    return false if recipients.nil?
+
+    begin
+      max_chans = Gitlab.config.irker.max_channels
+    rescue Settingslogic::MissingSetting
+      max_chans = 3
+    end
+
+    chans = recipients.split(/\s+/)
+    if chans.count > max_chans
+      errors.add(:recipients, "are limited to #{max_chans}")
+      return false
+    end
+
+    begin
+      default_irc = Gitlab.config.irker.default_irc_uri
+    rescue Settingslogic::MissingSetting
+      default_irc = nil
+    else
+      default_irc += '/' unless default_irc[-1] == '/'
+    end
+
+    self.channels = chans.map { |c|
+      cnt = 0
+      url = nil
+
+      # Now try to parse the chan as a full URI
+      begin
+        uri = URI.parse(c)
+        raise URI::InvalidURIError if uri.scheme.nil? and cnt == 0
+      rescue URI::InvalidURIError
+        unless default_irc.nil?
+          cnt += 1
+          c = "#{default_irc}#{c}"
+          retry if cnt == 1
+        end
+      else
+        # Authorize both irc://domain.com/#chan and irc://domain.com/chan
+        if uri.is_a?(URI) and uri.scheme[/^ircs?$/] and not uri.path.nil?
+          if uri.fragment.nil?
+            # Do not authorize irc://domain.com/
+            url = uri.to_s if uri.path.length > 1
+          else
+            # Authorize irc://domain.com/smthg#chan
+            # The irker daemon will deal with it by concatenating smthg and
+            # chan, thus sending messages on #smthgchan
+            url = uri.to_s
+          end
+        end
+      end
+      url
+    }.reject { |c| c.nil? }
+
+    errors.add(:recipients, "are all invalid") if self.channels.empty?
```

Comment:
Redundant `self` detected.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.